redLayerModule.py

```diff
@@ -299,7 +301,12 @@ def initGui(self):
             callback=self.loadAction,
             parent=self.iface.mainWindow(),
             object_name='mLoadSketchesFromFile')
-        self.canvasButton.setMenu(self.canvasMenu())
+        # create explicit QToolButton for canvas action (Qt5 & Qt6)
+        canvas_toolbutton = QToolButton(self.iface.mainWindow())
+        canvas_toolbutton.setDefaultAction(self.canvasButton)
+        canvas_toolbutton.setMenu(self.canvasMenu())
```

Comment:
```suggestion
        canvas_toolbutton.setMenu(self.canvasMenu())
        canvas_toolbutton.setPopupMode(QToolButton.InstantPopup)
```

It actually needs a popup mode, at least in QGIS3/Qt5. Otherwise the menu won't appear.